lib/adjust_tags/middleware.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+# Common space for Insights API stuff
+module Insights
+  module API
+    module Common
+      module AdjustTags
+        # Rack middleware to adjust params[tags] to be an array
+        class Middleware
+          def initialize(app)
+            @app = app
+          end
+
+          def call(env)
+            qs = env['QUERY_STRING'].sub(/^tags=/, 'tags[]=')
+                                    .gsub(/&tags=/, '&tags[]=')
+
+            env['REQUEST_URI'] = env['REQUEST_URI'].sub(env['QUERY_STRING'], qs)
```

Comment:
This could replace parts of URI that should not be replaced. Can it do the substitution from the end of the URI?